File: backend/schemas/collection.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field


# ── Folder schemas ──────────────────────────────────────────────────────────

class FolderCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=255)
    collection_id: str
    parent_folder_id: Optional[str] = None
    order_idx: int = 0


class FolderUpdate(BaseModel):
    name: Optional[str] = Field(None, min_length=1, max_length=255)
    parent_folder_id: Optional[str] = None
    order_idx: Optional[int] = None


class FolderResponse(BaseModel):
    id: str
    collection_id: str
    parent_folder_id: Optional[str]
    name: str
    order_idx: int
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


# ── Collection schemas ───────────────────────────────────────────────────────

class CollectionCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=255)
    description: Optional[str] = None


class CollectionUpdate(BaseModel):
    name: Optional[str] = Field(None, min_length=1, max_length=255)
    description: Optional[str] = None


class CollectionResponse(BaseModel):
    id: str
    name: str
    description: Optional[str]
    created_at: datetime
    updated_at: datetime
    folders: list[FolderResponse] = []

    model_config = {"from_attributes": True}

File: frontend/types/collection.ts
import type { RequestItem } from './request';

export interface Folder {
  id: string;
  collection_id: string;
  parent_folder_id: string | null;
  name: string;
  order_idx: number;
  created_at: string;
  updated_at: string;
}

export interface Collection {
  id: string;
  name: string;
  description: string | null;
  created_at: string;
  updated_at: string;
  folders: Folder[];
}

export interface CollectionWithRequests extends Collection {
  requests: RequestItem[];
}

export interface CollectionCreate {
  name: string;
  description?: string | null;
}

export interface CollectionUpdate {
  name?: string;
  description?: string | null;
}

export interface FolderCreate {
  name: string;
  collection_id: string;
  parent_folder_id?: string | null;
  order_idx?: number;
}

export interface FolderUpdate {
  name?: string;
  parent_folder_id?: string | null;
  order_idx?: number;
}
